src/apps/material-search/useGetSelectedWork.ts

```diff
@@ -66,6 +70,12 @@ const useGetSelectedWork = ({
     }
   );
 
+  useEffect(() => {
+    if (selectedWorkId) {
+      refetch();
+    }
+  }, [selectedWorkId, selectedMaterialType, refetch]);
```

Comment:
As I see it this is a problem because we unmount and remount the material search application in combination with the `onSuccess()` handler.

Another approach to do this would be to [clear the react query cache](https://tanstack.com/query/latest/docs/reference/QueryCache/#querycacheclear) [when applications are unmounted](https://github.com/danskernesdigitalebibliotek/dpl-react/blob/309df6a631653e3eee92414181cb52e3bb51245f/src/core/mount.js#L63-L63). This would ensure that they get a fresh state if they are remounted.